File: samples_generated_code/sample_3/qwen/baseline_code_c/code_row_93.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    char* name;
} Request;

typedef struct {
    Request* request;
    char* template;
} Environment;

char* get_name_parameter(Request* request) {
    return request->name;
}

char* render_template(char* template, char* name) {
    char* rendered = malloc(strlen(template) + strlen(name) + 1);
    sprintf(rendered, template, name);
    return rendered;
}

char* render_response_from_env(Environment* env) {
    char* name = get_name_parameter(env->request);
    char* rendered_template = render_template(env->template, name);
    return rendered_template;
}

char* home(Environment* env) {
    return render_response_from_env(env);
}

int main() {
    Request request = {"John"};
    Environment env = {&request, "Hello, %s!"};
    
    char* response = home(&env);
    printf("%s\n", response);
    
    free(response);
    return 0;
}

File: samples_generated_code/sample_3/athene/baseline_code_java/code_row_93.java
import java.util.Map;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

public class TemplateRenderer {

    public String render_response_from_env(Map<String, Object> environment, HttpServletRequest request) {
        String name = request.getParameter("name");
        String template = (String) environment.get("template");
        return template.replace("{name}", name);
    }

    public String home(Map<String, Object> env, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String renderedTemplate = render_response_from_env(env, request);
        response.setContentType("text/html");
        response.getWriter().write(renderedTemplate);
        return renderedTemplate;
    }
}